lib/components/data/tables/CWMSTable.jsx

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState, useRef, createRef } from "react";
+import { Configuration, TimeSeriesApi } from "cwmsjs";
+import dayjs from "dayjs";
+import { Table, TableBody, TableRow, TableHead, TableHeader, TableCell } from "@usace/groundwork";
+
+
+const config_v2 = new Configuration({
+    headers: {
+        accept: "application/json;version=2",
+    },
+});
+const ts_api = new TimeSeriesApi(config_v2);
+
+const pageSize = 500
+
+export default function CWMSTable({
+    timeseriesParams,
+    office,
+    begin,
+    end,
+    interval,
+    order,
+    unit = "EN",
+    missingString,
+    className = "",
+    responsive = true,
+    loadingComponent = null,
+}) {
+
+    const tableElement = useRef([]);
+    const [tableData, setTableData] = useState(null);
+
+    const [tsData, setTsData] = useState(null);
+
+    const table = []
+
+    useEffect(() => {
+
+        let tsids = []
+        Object.entries(timeseriesParams).map(item => tsids.push(item[1].tsid))
+
+        if (!tsids.length)
+            throw Error("You must specify one or more Timeseries IDs to table.");
+
+        if (!office) throw Error("You must specify a 3 letter ID for the office");
+
+        const fetchData = async () => {
+            let promises = tsids.map(async (name) => {
+                try {
+                    return await ts_api.getCwmsDataTimeseries({
+                        name,
+                        begin,
+                        end,
+                        pageSize,
+                        unit,
+                        office,
+                    });
+                } catch (error) {
+                    if (error.response?.status === 404) {
+                        console.warn(`Data for ${name} not found: 404`);
+                        return null;
+                    } else {
+                        throw error;
+                    }
+                }
+            });
+
+            let values = await Promise.all(promises);
+            let _data = { ts: {}, dates: [] };
+            let dates = []
+            if (!interval) { interval = 1 }
+
+            values.forEach((result) => {
+                if (result && result.values) {
+                    if (!_data.ts[result.values]) {
+                        _data.ts[result.name] = [];
+                    }
+                    _data.ts[result.name].push(result);
+                    result.values.forEach((item) => {
+                        const dt = item[0]
+                        const val = item[1]
+                        if (!dates.includes(dt) && val != null) {
+                            if (dt % (interval * 1000 * 60) == 0) {
+                                dates.push(dt)
+                            }
+                        }
+                    })
+                } else if (result === null) {
+                    console.warn(`Skipping as no data was found.`);
```

Comment:
If this happens for a "middle" time series, the column is still created but the values for the next series are put under the header for the missing data, e.g. if you create a table of outflows and elevations but the outflows are missing then the elevation values get listed under "outflows".  Need to make sure the table populates properly in this case.